### ISSUE TITLE
UptimeTimer::incrementElapsedTime() seems to have data races

### ISSUE THREAD

**Issue by MarkusTeufelberger** [ripple/rippled#2487]
TSAN complains and I kinda agree - there should probably be a lock/mutex somewhere here:

https://github.com/ripple/rippled/blob/1eece9b1fdbf58b8a4b52b738f53be4fd39dd37e/src/ripple/basics/impl/UptimeTimer.cpp#L69-L73

UptimeTimer in general looks fishy and hasn't been touched in years...

**Comment by nbougalis** [comment]
First, I agree that the class is old and junky and ought to be replaced. Paging @HowardHinnant...

Re: race condition: `UptimeTimer::incrementElapsedTime` is only ever called from a single thread, so it can't possibly race and the read of `m_elapsedTime` is protected by a fence and I don't think there are any atomicity/race issues. Beyond that, for technical reasons involving the width and operation of memory buses, operations on naturally aligned native type should be effectively atomic in all the modern CPU architectures I can think of.

**Comment by MarkusTeufelberger** [comment]
I guess a lot of errors come from the multithreaded test runner implementation. I also already saw some errors when initializing OpenSSLContext or whatever that ws called which also looked a bit like several threads being spawned for tests that wouldn't be allowed that way in normal operation.

**Comment by HowardHinnant** [comment]
I've been looking at this.  After good suggestions from Nik I'm going to create a chrono-style clock that measures uptime to seconds precision and has its own updater thread to increment the seconds.  This will mean that LoadManager is no longer responsible for manually updating the seconds count.

I'll also be swithcing from the fence to an atomic count to assure the correct multithreading semantics, but without the expense of locking a mutex per each read (which is occurring upwards of 10,000 times/sec).

Here's a demo of what this is looking like so far:
```
// header

#include <atomic>
#include <chrono>
#include <thread>

class UptimeClock
{
public:
    using rep        = int;
    using period     = std::ratio<1>;
    using duration   = std::chrono::duration<rep, period>;
    using time_point = std::chrono::time_point<UptimeClock>;
    static constexpr bool is_steady = false;

    explicit UptimeClock() = default;

    static time_point now();

private:
    static std::atomic<rep> now_;
    static std::atomic<bool> stop_;

    struct update_thread
        : private std::thread
    {
        ~update_thread();
        update_thread(update_thread&&) = default;

        using std::thread::thread;
    };

    static update_thread start_clock();
};

// source

std::atomic<UptimeClock::rep>  UptimeClock::now_{0};
std::atomic<bool>              UptimeClock::stop_{false};

UptimeClock::update_thread::~update_thread()
{
    if (joinable())
    {
        stop_ = true;
        join();
    }
}

UptimeClock::update_thread
UptimeClock::start_clock()
{
    return update_thread{[]
                         {
                             using namespace std;
                             using namespace std::chrono;
                             auto next = system_clock::now() + 1s;
                             while (!stop_)
                             {
                                 this_thread::sleep_until(next);
                                 next += 1s;
                                 ++now_;
                             }
                         }};
}

UptimeClock::time_point
UptimeClock::now()
{
    static const auto init = start_clock();
    return time_point{duration{now_.load()}};
}

// demo

#include <iostream>

int
main()
{
    using namespace std::chrono;
    auto t = UptimeClock::now();
    for (int i = 0; i < 10; ++i)
    {
        std::this_thread::sleep_for(500ms);
        t = UptimeClock::now();
        std::cerr << t.time_since_epoch().count() << '\n';
    }
}
```

**Comment by nbougalis** [comment]
Addressed in #2492 which is pending. Closing this issue.
